teachertool/src/components/CriteriaResultEntry.tsx

```diff
@@ -65,61 +44,90 @@ const CriteriaResultNotes: React.FC<CriteriaResultNotesProps> = ({ criteriaId })
+    const hasFeedback = !!evalResult?.notes;
 
     const criteriaInstance = getCriteriaInstanceWithId(teacherTool, criteriaId);
     const catalogCriteria = criteriaInstance ? getCatalogCriteriaWithId(criteriaInstance.catalogCriteriaId) : undefined;
 
-    useEffect(() => {
-        if (!showInput && teacherTool.evalResults[criteriaId]?.notes) {
-            setShowInput(true);
-        }
-    }, [teacherTool.evalResults[criteriaId]?.notes]);
-
-    const isInProgress = teacherTool.evalResults[criteriaId].result === EvaluationStatus.InProgress;
+    const isInProgress = evalStatus === EvaluationStatus.InProgress;
     return (
         <>
-            {catalogCriteria && (
+            {catalogCriteria && criteriaInstance && (
                 <div className={css["specific-criteria-result"]} key={criteriaId} aria-busy={isInProgress}>
                     {/* Criteria Text & Overall Result (Looks Good, Needs Work, Etc...) */}
                     <div className={css["result-details"]}>
-                        <ReadOnlyCriteriaDisplay
-                            catalogCriteria={catalogCriteria}
-                            criteriaInstance={criteriaInstance}
-                            showDescription={false}
-                        />
+                        <CriteriaInstanceDisplay criteriaInstance={criteriaInstance} />
                         {!isInProgress && (
-                            <CriteriaEvalResultDropdown
-                                result={teacherTool.evalResults[criteriaId].result}
-                                criteriaId={criteriaId}
-                            />
+                            <div className={css["result-controls"]}>
+                                <CriteriaEvalResultDropdown result={evalStatus} criteriaId={criteriaId} />
+                                <CriteriaResultToolbarTray criteriaId={criteriaId} />
+                            </div>
                         )}
                     </div>
 
                     {/* Notes */}
                     {isInProgress ? (
                         <ThreeDotsLoadingDisplay className={css["loading-display"]} />
                     ) : (
-                        <div
-                            className={classList(
-                                css["result-notes"],
-                                teacherTool.evalResults[criteriaId]?.notes ? undefined : css["no-print"]
-                            )}
-                        >
-                            {!showInput && <AddNotesButton criteriaId={criteriaId} setShowInput={setShowInput} />}
-                            {showInput && <CriteriaResultNotes criteriaId={criteriaId} />}
+                        <div className={classList(css["result-notes"], !hasFeedback ? "no-print" : undefined)}>
+                            <CriteriaResultNotes criteriaId={criteriaId} />
                         </div>
                     )}
 
+                    <div className={css["separator"]}></div>
```

Comment:
Should this come after the CriteriaFeedback below? As is, it ends up looking like the thumbs up and down are at the top of the following feedback instead of the bottom of the current feedback:
![image](https://github.com/user-attachments/assets/e658942e-56bf-4b86-9cd9-3d24f83862b9)

Reply:
good catch!